src/main/host/syscall/handler/mod.rs

```diff
@@ -319,7 +321,25 @@ impl SyscallHandler {
             // UNSUPPORTED SYSCALL
             //
             _ => {
-                log::warn!("Detected unsupported syscall {} ({}) called from thread {} in process {} on host {}",
+                // only show a warning the first time we encounter this unsupported syscall
+                static WARNED_SET: Mutex<Option<HashSet<SyscallNum>>> = Mutex::new(None);
+
+                // `insert()` returns `false` if the syscall num was already in the set
+                let has_already_warned = !WARNED_SET
+                    .lock()
+                    .unwrap()
+                    .get_or_insert_with(HashSet::new)
+                    .insert(syscall);
+
+                let level = if has_already_warned {
+                    log::Level::Debug
+                } else {
+                    log::Level::Warn
+                };
+
+                log::log!(
```

Comment:
Maybe add a comment here saying that we can't use the `warn_once_then_debug` macro because that warns once per specific log message whereas here we need `warn_once` behavior based on the syscall number not the log message itself.

Reply:
I'll add this. And just to clarify, it logs once per line of code (or more specifically, once per macro call), not per specific log message. So if you had `log_once_then_debug!("Foo {}", x)`, it would only ever log that line once, even if `x` changed.